eZ/Publish/API/Repository/ContentService.php

```diff
@@ -432,4 +432,15 @@ public function newTranslationInfo();
      * @return \eZ\Publish\API\Repository\Values\Content\TranslationValues
      */
     public function newTranslationValues();
+
+    /**
+     * Returns true if the given content type $contentType has content instances.
+     *
+     * @since 6.0.1
+     *
+     * @param \eZ\Publish\API\Repository\Values\ContentType\ContentType $contentType
+     *
+     * @return bool
+     */
+    public function isContentTypeUsed(ContentType $contentType);
```

Comment:
i'd definitely go for the ```Section``` way. In ```ContentTypeService``` i'll add a isContentTypeUsed method like...
```php
public function isContentTypeUsed(ContentType $contentType)
{
     return $this->contentTypeHandler->assignmentsCount($contentType->id) > 0;
}
```
then, in then ContentTypeHandler...
```php
public function assignmentsCount($contentTypeId)
 {
     return $this->contentTypeGateway->countContentObjectsOfContentType($contentTypeId);
 }
```
and in the Doctrine Gateway we could a function very similar to https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Persistence/Legacy/Content/Section/Gateway/DoctrineDatabase.php#L184